### ISSUE TITLE
Writing an empty geodataframe to .geojsons results in 0 byte file

### ISSUE THREAD

**Issue by theroggy** [geopandas/pyogrio#94]
Probably an issue in gdal...

Can be reproduced in a test that will be added in #75: test_write_empty_dataframe_unsupported()

**Comment by brendan-ward** [comment]
I am not able to reproduce (MacOS 10.15, GDAL 3.4.1); pyogrio produces an empty GeoJSONSeq file (0 bytes).  This is identical to what I get from GeoPandas (via Fiona) using fixes in [geopandas #2240](https://github.com/geopandas/geopandas/pull/2240) to allow writing empty data frames and to what I get using `ogr2ogr` to convert an empty GeoJSON to GeoJSONSeq.

What version of GDAL are you using?

**Comment by jorisvandenbossche** [comment]
> pyogrio produces an empty GeoJSONSeq file (0 bytes)

This is also what the title of the issue says?

Although it is annoying that it cannot read such an empty file (that is maybe something to report to gdal?), I suppose this is the expected behaviour? Given that a geojsonseq file consists of one line per record, if there are no records there are no lines in the file?

**Comment by brendan-ward** [comment]
Ah yes, didn't read that, jumped from #101 where I was next going to comment along the lines of why are we attempting to read it when we should just check existence of it...  

I don't know what I would ask GDAL to do differently: not create a file at all?  That also seems to break the model with something like `ogr2ogr` where you would always expect an output file.

**Comment by jorisvandenbossche** [comment]
Yes, I don't think writing it is a problem, it's just that ogr2ogr also is not able to roundtrip, it can't read the file it has written. If you start from an empty geojson:

```
$ ogr2ogr empty.geojsons empty.geojson
$ ogr2ogr empty_roundtrip.geojson empty.geojsons
FAILURE:
Unable to open datasource `empty.geojsons' with the following drivers.
...
```

**Comment by theroggy** [comment]
Yes, indeed the way it is written is probably OK, but it seems more logical at first sight that gdal would be able to read an empty .geojsons file as a datasource with 0 rows...

**Comment by theroggy** [comment]
I update the title and description based on the new insights...
